### PR TITLE
feat: add config file loader

### PR DESCRIPTION
**Pull request by gu18168**

### CODE REVIEW

File: conf/conf.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+discovery:                       # service discovery center
+  nacos:
+    host:                        # it's possible to define multiple nacos hosts addresses of the same eureka cluster.
+      - "http://127.0.0.1:8848"
+    prefix: /nacos/v1/
+    weight: 100                  # default weight for node
+    timeout:
+      connect: 2000              # default 2000ms
+      send: 2000                 # default 2000ms
+      read: 5000                 # default 5000ms
```
Comment (spacewander) [inline conf/conf.yaml]: Let's configure your editor to add a newline at the end of file.

File: conf/conf.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+etcd:
+  host:                           # it's possible to define multiple etcd hosts addresses of the same etcd cluster.
+    - "http://127.0.0.1:2379"     # multiple etcd address, if your etcd cluster enables TLS, please use https scheme,
+    # e.g. https://127.0.0.1:2379.
+  prefix: /apisix                 # apisix configurations prefix
+  timeout: 30                     # 30 seconds
+  #resync_delay: 5                # when sync failed and a rest is needed, resync after the configured seconds plus 50% random jitter
```
Comment (spacewander) [inline conf/conf.yaml]: We can remove the unsupported fields instead of comment out them.

File: internal/utils/test_case.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (spacewander) [inline internal/utils/test_case.json]: Better to put the file under testdata/. See https://dave.cheney.net/2016/05/10/test-fixtures-in-go

File: conf/conf.yaml (hunk 1 above)
Comment (gu18168) [reply conf/conf.yaml]: Done.

File: internal/utils/test_case.json (hunk 3 above)
Comment (gu18168) [reply internal/utils/test_case.json]: I have reorganized the additional files needed for the test.

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
Fantasy!
Let's set up the unit test and lint like these:
https://github.com/apache/apisix-go-plugin-runner/blob/master/.github/workflows/unit-test-ci.yml
https://github.com/apache/apisix-go-plugin-runner/blob/master/.github/workflows/lint.yml

**Pull request merged**